### PR TITLE
Add new command cylance-protect-get-device-by-hostname

### PR DESCRIPTION
**Pull request by PaulBartruff**
## Status
- [] In Progress
- [X ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes:
Fully: https://github.com/demisto/etc/issues/37985
Partially: https://github.com/demisto/etc/issues/16615
Invalidates: https://github.com/demisto/etc/issues/23512

## Description
Recent Cylance Protect API guides provide additional features, such as getting a device by hostname vs knowing it UID ahead of time.  The existing integration requires iterating through all devices to find the device one is searching for.  Using the newer API, one can shortcut the iteration and just get the device via the hostname.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [ ] 5.5.0
- [ ] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by pbarttruff** [comment]
I was not sure how to add the new command to the README.md  I tried the demisto-sdk; however, I was not confident in the output of the utility :-P

File: Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.yml [hunk 1]
```diff
@@ -172,74 +163,104 @@ script:
...
+      type: Unknown
+    - contextPath: Endpoint.OSVersion
+      description: Device OS version.
+      type: string
+    - contextPath: CylanceProtect.Device.OSVersion
+      description: Device OS version.
+      type: string
+    - contextPath: CylanceProtect.Device.Name
+      description: Device name.
+      type: string
```
Comment (tomneeman151293) [inline Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.yml:229]: Seems you had AgentVersion output which was not really outputted. Also you haven't added Policy outputs to Device. I have added them here, double check to see if anything is missing.
Also, added TODO for what I added for description and type. Please fill
adding a suggestion, let me know if I missed something and it was configured correctly
```suggestion
    - contextPath: CylanceProtect.Device.DateFirstRegistered
      description: The date and time (in UTC) when the device record was created.
      type: date
    - contextPath: CylanceProtect.Device.BackgroundDetection
      description: If true, the Agent is currently running.
      type: boolean
    - contextPath: CylanceProtect.Device.DateLastModified
      description: The date and time (in UTC) when the device record was last modified.
      type: date
    - contextPath: CylanceProtect.Device.DateOffline
      description: The date and time (in UTC) when the device last communicated with
        the Console.
      type: date
    - contextPath: CylanceProtect.Device.Hostname
      description: The hostname for the device.
      type: string
    - contextPath: CylanceProtect.Device.ID
      description: The unique identifier for the device.
      type: string
    - contextPath: CylanceProtect.Device.IPAddress
      description: The list of IP addresses for the device.
      type: Unknown
    - contextPath: CylanceProtect.Device.MACAddress
      description: The list of MAC addresses for the device.
      type: Unknown
    - contextPath: CylanceProtect.Device.IsSafe
      description: If true, there are no outstanding threats.
      type: boolean
    - contextPath: CylanceProtect.Device.UpdateAvailable
      description: If true, there is available update for the device.
      type: boolean
    - contextPath: CylanceProtect.Device.State
      description: Machine state.
      type: string
    - contextPath: Endpoint.Hostname
      description: Device hostname.
      type: string
    - contextPath: Endpoint.MACAddress
      description: The list of MAC addresses for the device.
      type: Unknown
    - contextPath: Endpoint.IPAddress
      description: The list of IP addresses for the device.
      type: Unknown
    - contextPath: Endpoint.OSVersion
      description: Device OS version.
      type: string
    - contextPath: CylanceProtect.Device.OSVersion
      description: Device OS version.
      type: string
    - contextPath: CylanceProtect.Device.Name
      description: Device name.
      type: string
    - contextPath: CylanceProtect.Device. LastLoggedInUser
      description: TODO add description.
      type: TODO  add the type
    - contextPath: CylanceProtect.Device.UpdateType
      description: TODO add description.
      type: TODO  add the type
    - contextPath: CylanceProtect.Device.Policy.ID
      description: TODO add description.
      type: TODO  add the type
    - contextPath: CylanceProtect.Device.Policy.Name
      description: TODO add description.
      type: TODO  add the type
```

**Review by tomneeman151293 — CHANGES_REQUESTED** [review]
Looks very nice! Thanks for the contribution
Please:
1) run demisto-sdk format -i Packs/Cylance_Protect/Integrations/Cylance_Protect_v2/Cylance_Protect_v2.yml
2) remove fromversion field from Cylance_Protect_v2.yml after running 1
3) demisto-sdk update-release-notes -i Packs/Cylance_Protect -u revision
4) Number 3 will generate Release Notes file in the pack. Please add the entry
- Added the **cylance-protect-get-device-by-hostname** command
5) Please add unit tests for the added command.

After those, lets arrange a short demo to see the newly added command

**Comment by tomneeman151293** [comment]
I will fix validations for your branch and push the changes

**Comment by PaulBartruff** [comment]
@tomneeman151293 Awesome! That works for me.  I'll review the api release notes and history to check when the command was added.  Also, I tested this with our version of Cylance and a customer's which was version: 2.1.1574

R/
Paul

**Comment by PaulBartruff** [comment]
@tomneeman151293
I tried to run down the publicly accessible API docs to compare when the command became supported.  Here is what I got:

Cylance API Release Notes V1.1.0 Dated: 2020-11-06
[API v1.10](https://docs.blackberry.com/content/dam/docs-blackberry-com/release-pdfs/en/cylance-api/Cylance-API-1.1.0-release-notes.pdf)

Had the "Get Device by Hostname" command; however, a fix was issued to make the command return similar results as other "Get" commands.

Cylance API User API v2.0 rev34 Guide Dated: December 2020
[API v2.0 rev34](https://docs.blackberry.com/content/dam/docs-blackberry-com/release-pdfs/en/cylance-products/api-and-developer-guides/Cylance%20User%20API%20Guide%20v2.0%20rev34.pdf)

This is the guide I used to make the change.  I do not have any other data points to validate previous versions.
R/
Paul

**Comment by tomneeman151293** [comment]
Ok, I will contact you later today to schedule a quick demo for the command

**Comment by tomneeman151293** [comment]
@PaulBartruff Thanks a lot for your demo. I am approving and adding this to our content

**Review by tomneeman151293 — APPROVED** [review]

**Pull request merged**